diffrax/_solver/rosenbrock.py

```diff
@@ -0,0 +1,247 @@
+    first array represents the should be of shape `(1,)`. Each subsequent array should
+    be of shape `(2,)`, `(3,)` etc. The final array should have shape `(k - 1,)`.
+- `c_lower`: the lower triangle (without the diagonal) of the tableau. Should
+    be a tuple of NumPy arrays, corresponding to the rows of this lower triangle. The
+    first array represents the should be of shape `(1,)`. Each subsequent array should
+    be of shape `(2,)`, `(3,)` etc. The final array should have shape `(k - 1,)`.
+- `m_sol`: the linear combination of stages to take to produce the output at each step.
+    Should be a NumPy array of shape `(k,)`.
+- `m_error`: the linear combination of stages to produce a lower-order solution
+    for error estimation. Should be a NumPy array of shape `(k,)`. The error is
+    calculated as the difference between the main solution (using `m_sol`) and
+    this lower-order solution (using `m_error`), providing an estimate of the
+    local truncation error for adaptive step size control.
+- `α`: the time increment.
+- `γ`: the vector field increment.
+"""
+
+
+class AbstractRosenbrock(AbstractAdaptiveSolver):
+    r"""Abstract base class for Rosenbrock solvers for stiff equations.
+
+    Subclasses should define `tableau` and `interpolation_cls` as class-level attributes
+    `tableau` should be an instance of `diffrax.RosenbrockTableau`, and
+    `interpolation_cls` should be an instance of `diffrax.AbstractLocalInterpolation`.
+    """
+
+    term_structure: ClassVar = AbstractTerm
+
+    tableau: ClassVar[RosenbrockTableau]
+
+    rodas: ClassVar[bool] = False
+
+    linear_solver: lx.AbstractLinearSolver = lx.AutoLinearSolver(well_posed=True)
+
+    def init(self, terms, t0, t1, y0, args) -> _SolverState:
+        del t0, t1
+        if any(
+            eqx.is_array_like(xi) and jnp.iscomplexobj(xi)
+            for xi in jtu.tree_leaves((terms, y0, args))
+        ):
+            # TODO: add complex dtype support.
+            raise ValueError("rosenbrock does not support complex dtypes.")
+
+    def step(
+        self,
+        terms: AbstractTerm,
+        t0: RealScalarLike,
+        t1: RealScalarLike,
+        y0: Y,
+        args: Args,
+        solver_state: _SolverState,
+        made_jump: BoolScalarLike,
+    ) -> tuple[Y, Y, DenseInfo, _SolverState, RESULTS]:
+        del solver_state, made_jump
+
+        y0_leaves = jtu.tree_leaves(y0)
+        sol_dtype = jnp.result_type(*y0_leaves)
+
+        control = terms.contr(t0, t1)
+        identity = jtu.tree_map(lambda leaf: jnp.ones_like(leaf), control)
+
+        time_derivative = jax.jacfwd(lambda t: terms.vf_prod(t, y0, args, identity))(t0)
+        time_derivative, unravel_t = fu.ravel_pytree(time_derivative)
+
+        jacobian = jax.jacfwd(lambda y: terms.vf_prod(t0, y, args, identity))(y0)
+        jacobian, _ = fu.ravel_pytree(jacobian)
+        jacobian = jnp.reshape(jacobian, time_derivative.shape * 2)
```

Comment:
I had to materialize the jacobian, as it is used in rhs as well in the every stage equation.